src/ucp/api/ucp.h

```diff
@@ -2372,6 +2474,33 @@ ucs_status_t ucp_worker_set_am_handler(ucp_worker_h worker, uint16_t id,
                                        uint32_t flags);
 
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Add user defined callback for Active Message.
+ *
+ * This routine installs a user defined callback to handle incoming Active
+ * Messages with a specific id. This callback is called whenever an Active
+ * Message that was sent from the remote peer by @ref ucp_am_send_nbx is
+ * received on this worker.
+ *
+ * @warning Handlers set by this function are not compatible with
+            @ref ucp_am_send_nb routine.
+ *
+ * @param [in]  worker      UCP worker on which to set the Active Message
+ *                          handler.
+ * @param [in]  id          Active Message id.
+ * @param [in]  cb          Active Message callback. NULL to clear.
+ * @param [in]  param       Active Message handler parameters, as defined by
+ *                          @ref ucp_am_handler_param_t.
+ *
+ * @return error code if the worker does not support Active Messages or
+ *         requested callback flags.
+ */
+ucs_status_t ucp_worker_set_am_data_handler(ucp_worker_h worker, uint16_t id,
```

Comment:
does it replace ucp_worker_set_am_handler?

Reply:
no, we keep existing API unchanged